File: src/Visit.java
import java.util.ArrayList;

public class Visit {

    private ArrayList<Visitor> visitors;
    private double totalCost;
    private String dateVisit;
    private Employee guide;
    private int numberChildren=0;
    private int numberAdults=0;

    public Visit(Employee employee, Visitor visitor, String dateVisit) {
        this.guide = employee;
        this.visitors = new ArrayList<>(); // Inicializamos el ArrayList
        this.visitors.add(visitor);
        visitor.addVisit();
        this.dateVisit = dateVisit;
        this.totalCost=0;

    }

    public int getNumberPeople(){
        return visitors.size();
    }

    public void addVisitor(Visitor visitor){
        boolean band=false;
        for (Visitor visitor1: visitors){
            if (visitor1==visitor){
                System.out.println("VISITOR ALREADY EXISTS IN THE VISIT.");
                band=true;
                break;
            }
        }
        if (!band){
            this.visitors.add(visitor);
            visitor.setVisitNumber(1);
        }

    }

    public double calculateTotalPrice(){
        for (Visitor visitor: visitors){
            int total=visitor.getVisitNumber();
            if (total==5){
                if (visitor.checkAdult()){
                    totalCost+= (float) (100.0*0.8);
                } else{
                    totalCost+= (float) (50.0*0.8);
                }
                visitor.setVisitNumber(0);
            } else{
                if (visitor.checkAdult()){
                    totalCost+= 100.0;
                } else{
                    totalCost+= 50.0;
                }
            }
        }
        return totalCost;
    }

    public String getDateVisit() {
        return dateVisit;
    }

    public void setDateVisit(String dateVisit) {
        this.dateVisit = dateVisit;
    }

    public Employee getGuide() {
        return guide;
    }

    public void setGuide(Employee guide) {
        this.guide = guide;
    }

    public int getNumberChildren() {
        for (Visitor visitor: visitors){
            boolean band=false;
            band = visitor.checkAdult();
            if (!band){
                this.numberChildren+=1;
            }
        }
        return numberChildren;
    }

    public void setNumberChildren(int numberChildren) {
        this.numberChildren = numberChildren;
    }

    public int getNumberAdults() {
        for (Visitor visitor: visitors){
            boolean band=false;
            band = visitor.checkAdult();
            if (band){
                this.numberAdults+=1;
            }
        }
        return numberAdults;
    }

    public void setNumberAdults(int numberAdults) {
        this.numberAdults = numberAdults;
    }
    public ArrayList<Visitor> getVisitors(){
        return visitors;
    }
}

File: src/Visit.py
from Employee import Employee


class Visit:
    __total_cost = 0
    __date_visit = ""
    __guide = Employee
    __number_children = 0
    __number_adults = 0

    def __init__(self, employee, visitor, date_visit):
        self.__guide = employee
        self.__visitors = []
        self.__visitors.append(visitor)
        self.__date_visit = date_visit
        self.__total_cost = 0

    def get_number_people(self):
        return len(self.__visitors)

    def add_visitor(self, visitorsel):
        band = False
        for visitor in self.__visitors:
            if visitor == visitorsel:
                print("Visitor already exists in the visit. ")
                band = True
                break
        if not band:
            self.__visitors.append(visitorsel)
            visitorsel.set_visit_number(1)

    def calculate_total_price(self):
        for visitor in self.__visitors:
            total = visitor.get_visit_number()
            if total == 5:
                if visitor.check_adult():
                    self.__total_cost += 100.0 * 0.8
                else:
                    self.__total_cost += 50.0 * 0.8
                visitor.set_visit_number(0)
            else:
                if visitor.check_adult():
                    self.__total_cost += 100.0
                else:
                    self.__total_cost += 50.0
        return self.__total_cost

    def get_date_visit(self):
        return self.__date_visit

    def get_guide(self):
        return self.__guide

    def set_guide(self, guide):
        self.__guide = guide

    def get_number_children(self):
        for visitor in self.__visitors:
            band = visitor.check_adult()
            if not band:
                self.__number_children += 1

        return self.__number_children

    def set_number_children(self, number_children):
        self.__number_children = number_children

    def get_number_adults(self):
        for visitor in self.__visitors:
            band = visitor.check_adult()
            if band:
                self.__number_adults += 1

        return self.__number_adults

    def set_number_adults(self, number_adults):
        self.__number_adults = number_adults

    def get_visitors(self):
        return self.__visitors
